lib/puppet/provider/elastic_rest.rb

```diff
@@ -33,7 +33,7 @@ def self.rest(http,
                 timeout = 10,
                 username = nil,
                 password = nil,
-                validate_tls: true)
+                validate_tls = true)
```

Comment:
This change the semantic and is probably not what you want (it seems to not break tests so this demonstrate that unit tests are missing).

You can read more about "keyword arguments" here if some context is needed:
https://www.rubyguides.com/2018/06/rubys-method-arguments/

For now, I would suggest adding a unit test that demonstrate that the current behavior is wrong in a first commit, and then fix it with this change (2 commits in the PR).

Reply:
i rollback